src/uwtools/drivers/driver.py

```diff
@@ -53,10 +58,36 @@ class Driver(ABC):
     run and will appropriately parse subsequent stages of the workflow.
     """
 
-    def __init__(self):
+    def __init__(self, config_file: Optional[str] = None):
         """
         Initialize the Forecast driver.
         """
+        if config_file is not None:
+            self.config_file = config_file
+            self.validate()
+
+    def validate(self, log: Optional[Logger] = None) -> bool:
+        """
+        Validates the objects supplied to the driver from the provided config and platform files.
+        """
+
+        log = Logger() if log is None else log
+        return (
+            config_validator.config_is_valid(
+                config_file=self.config_file,
+                validation_schema=self.schema,
+                log=log,
+            )
+            if self.config_file and self.schema
+            else sys.exit(1)
+        )
```

Comment:
I'm still catching up on my emails and haven't seen yet how this is going to be tested, but in case @WeirAE is working on that now, I'd like to propose that, instead of doing this check and potentially exiting here, we should just make `validate()` -> `_validate()` (private). The constructor already makes sure that this method is only called when `config_file` is provided, so we shouldn't **be** here if it wasn't. Also, a method on an object should not take down the entire Python application unless there's a very serious issue!

Reply:
In general, it's also good design to avoid side effects (e.g. killing Python) wherever possible. From best to worst error-handling behavor:
1. A pure function that simply returns a value that can be evaluated by the caller, which can then decide what to do.
2. Raise an exception, which can at least be caught by higher-level code and handled if possible. If no code catches the exception, _then_ the application will die.
3. Killing the application.

Reply:
Sorry for the extra email, turns out I missed a save there! I do need to rework the `Driver` and test functions now, since we can't query `instance.validate()` anymore so it'll have to return a message. The full sys.exit(1) I actually pulled from the behavior of `validate_config.py`, so that may need to be updated as well

Reply:
Yes, I assumed there were other places where we're not doing this as well as we could that we should improve, but I figured we could at least not add another. Thanks for addressing it.

In a well-designed code, all functions/methods would just propagate values (or, worst case, exceptions) back up the call stack -- logging their complaints, of course, because they've got the local knowledge of what happened -- until the outermost CLI level, which would then call `exit()` with either `0` or `1` to denote success or failure. (Or other non-`0` values if it's really useful to distinguish different failure reasons numerically -- this was old-school Unix behavior, but a lot of modern software sticks with `0` and `1`.)

For motivation, imagine that you're running different activities inside your program concurrently, in threads or futures. If one thread calls `exit()`, the entire Python interpreter dies, and all the other threads die in the middle of whatever they were doing, which may leave a mess. But if a thread with an issue just exits nicely with a return value (or even an exception), the main thread that spawned all the other worker threads can detect that and decide what to do. It might wait until all the other threads return on their own, letting them complete their work, clean up after themselves, etc.; or it might decide that the situation is so dire that it has to kill all those threads mid-work anyway. But that decision should be made at a fairly high level. This also makes all those functions/methods easier to test, because they don't have to test for the "else exit(1)` branches!

Reply:
Replaced this behavior with logging like a sensible person! The CLI should be making decisions about exiting with that information

Reply:
Looks good!